app/src/main/java/com/framgia/soundcloud/screen/listtracks/ListtracksViewModel.java

```diff
@@ -21,13 +25,34 @@
     private ListtrackAdapter mAdapter;
     private Context mContext;
     private boolean mIsLoading;
+    private String mUrl;
+    private RecyclerView.OnScrollListener mScrollListener = new RecyclerView.OnScrollListener() {
+        @Override
+        public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
+            super.onScrolled(recyclerView, dx, dy);
+            if (dy <= 0) {
+                return;
+            }
+            LinearLayoutManager layoutManager =
+                    (LinearLayoutManager) recyclerView.getLayoutManager();
+            int visibleItemCount = layoutManager.getChildCount();
+            int totalItemCount = layoutManager.getItemCount();
+            int pastVisibleItems = layoutManager.findFirstVisibleItemPosition();
+            if (!isLoading() && (visibleItemCount + pastVisibleItems) >= totalItemCount) {
+                setLoading(true);
+                mPresenter.LoadMoreTracks(mUrl);
+            }
+        }
+    };
 
     public ListtracksViewModel(Context context, ListtracksContract.Presenter presenter) {
         mPresenter = presenter;
         mPresenter.setViewModel(this);
         mPresenter.getTracks();
         mContext = context;
         mIsLoading = true;
+        sTracks = new ArrayList<>();
```

Comment:
ERROR: [STYLE][ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD] Write to static field com.framgia.soundcloud.screen.listtracks.ListtracksViewModel.sTracks from instance method new com.framgia.soundcloud.screen.listtracks.ListtracksViewModel(Context, ListtracksContract$Presenter)
http://findbugs.sourceforge.net/bugDescriptions.html#ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD